csrc/id_model/indexing.cpp

```diff
@@ -1125,9 +1125,20 @@ std::vector<PredicateInfo> TensorIndexer::getPredicates(
   // If this is a reduction init expr, then no need to take care of
   // non divisible splits
   if (!lower_utils::isReductionInitExpr(expr)) {
-    for (const PredicateDomainInfo& pred_info :
-         non_divisible_split_predicates) {
-      IterDomain* non_divisible_domain = pred_info.id;
+    for (const auto& [eg, direction] : index_info.traversal_path) {
```

Comment:
`pred_info` has all the non-divisible splits of the consumer tensor, but they may not be the ones that should be added here since the actual indexing path may not include them. See the added test. The broadcast tensor, `tv2`, has a non-divisible split, but the real domain that needs to be predicated is the one showing up in its indexing path, which is the split in the `tv3` tensor.